File: YeoHun/211216/숫자게임.swift
import Foundation

func solution(_ a:[Int], _ b:[Int]) -> Int {
    var sortedB = b.sorted(by: <)
    let sortedA = a.sorted(by: <)
    
    func binarySearch(target: Int) -> Int {
        var low = 0
        var high = sortedB.count - 1
        while low < high {
            let mid = (low + high) / 2
            if sortedB[mid] > target {
                high = mid
            } else {
                low = mid + 1
            }
        }
        return low
    }
    
    var answer = 0
    for card in sortedA {
        let removed = sortedB.remove(at: binarySearch(target: card))
        if card < removed {
            answer += 1
        }
    }
    
    return answer
}

File: TaeHoon/211216/숫자게임.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> A, vector<int> B) {
    int answer = 0;
    
    sort(A.begin(), A.end());
    sort(B.begin(), B.end());
    int i = 0;
    int j = 0;
    while (i < A.size() && j < B.size()) {
        while (A[i] >= B[j] && j < B.size())
            j++;
        if (j >= B.size())
            break;
        if (A[i] < B[j])
            answer++;
        j++;
        i++;
    }
    return answer;
}
